src/needsahome.jl

```diff
@@ -21,11 +21,11 @@ Related
 
 getVariableIds, _copyIntoGraph!
 """
-function buildSubgraphFromLabels!(dfg::G,
+function buildSubgraphFromLabels!(dfg::GSRC,
                                   syms::Vector{Symbol};
-                                  subfg::AbstractDFG=(G <: InMemoryDFGTypes ? G : GraphsDFG)(params=getSolverParams(dfg)),
+                                  subfg::GDST=(GSRC <: InMemoryDFGTypes ? GSRC : GraphsDFG)(params=getSolverParams(dfg)),
                                   solvable::Int=0,
-                                  allowedFactors::Union{Nothing, Vector{Symbol}}=nothing  )::G where G <: AbstractDFG
+                                  allowedFactors::Union{Nothing, Vector{Symbol}}=nothing  )::GDST where {GSRC <: AbstractDFG, GDST <: AbstractDFG}
```

Comment:
When this function was in IIF, the idea behind it was that in-memory graph types should not be converted between types and that other types would default to the graph defined in `const InMemDFGType`
I'm a bit worried that CGDFG might be broken.
It is probably a good idea to allow any` subfg <: InMemoryDFGTypes`, but this new definition does not make much sense. I will investigate.

Reply:
Perhaps this fits in with the new talking point at #264